discovery-infra/test_infra/kubeapi_utils.py

```diff
@@ -91,3 +98,48 @@ def get_hostname_for_agent(agent, nodes_details):
             if interface['macAddress'].lower() == mac_address:
                 return node_metadata['name']
 
+
+def delete_kube_api_resources_for_namespace(
+        kube_api_client,
+        name,
+        namespace,
+        *,
+        secret_name=None,
+        installenv_name=None,
+        nmstate_name=None,
+):
+    CoreV1Api.delete_namespaced_secret = suppress_not_found_error(
+        fn=CoreV1Api.delete_namespaced_secret,
+    )
+    CustomObjectsApi.delete_namespaced_custom_object = suppress_not_found_error(
+        fn=CustomObjectsApi.delete_namespaced_custom_object
+    )
+
+    cluster_deployment = ClusterDeployment(
+        kube_api_client=kube_api_client,
+        name=name,
+        namespace=namespace,
+    )
+
+    for agent in cluster_deployment.list_agents():
+        agent.delete()
+
+    cluster_deployment.delete()
+
+    Secret(
```

Comment:
don't we want to delete maximum we can? i think we should try except every delete operation or there is no way that it will fail?

Reply:
this is basically what I've done here, the only acceptable exception is NotFound, otherwise, this is probably an error we want to investigate no?

```
    CoreV1Api.delete_namespaced_secret = suppress_not_found_error(
        fn=CoreV1Api.delete_namespaced_secret,
    )
    CustomObjectsApi.delete_namespaced_custom_object = suppress_not_found_error(
        fn=CustomObjectsApi.delete_namespaced_custom_object
    )
```